real_time/surface_src/java_src/Alice/src/test/java/com/roboclub/robobuggy/main/RobobuggyConfigFileTest.java

```diff
@@ -0,0 +1,216 @@
+        //Runs after each test case
+    }
+
+    public void createConfigFileWithRobot(String s) throws IOException{
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File("/Users/davidneiman/RoboBuggy/real_time/surface_src/java_src/Alice/config.config")));
+        writer.write("{\"COM_PORT_IMU\":\"NOT_SET\",\"COM_PORT_RBSM\":\"NOT_SET\",\"PORT_VISION\":\"NOT_SET\",\"COM_PORT_GPS\":\"NOT_SET\",\"WAYPOINT_SOURCE_LOG_FILE\":\"config.config\",\"PLAY_BACK_SOURCE_FILE\":\"NOT_SET\",\"VIDEO_FILE\":\"/Users/davidneiman/RoboBuggy/real_time/surface_src/java_src/Alice/logs/2016-02-20-06-50-45/webcam.mp4\",\"ROBOT_TYPE\":\"");
+        writer.write(s);
+        writer.write("\"}");
+        writer.close();
+    }
+
+    public void createConfigFileWithoutRobot() throws IOException{
+        BufferedWriter writer = new BufferedWriter(new FileWriter(new File("/Users/davidneiman/RoboBuggy/real_time/surface_src/java_src/Alice/config.config")));
+        writer.write("{\"COM_PORT_IMU\":\"NOT_SET\",\"COM_PORT_RBSM\":\"NOT_SET\",\"PORT_VISION\":\"NOT_SET\",\"COM_PORT_GPS\":\"NOT_SET\",\"WAYPOINT_SOURCE_LOG_FILE\":\"config.config\",\"PLAY_BACK_SOURCE_FILE\":\"NOT_SET\",\"VIDEO_FILE\":\"/Users/davidneiman/RoboBuggy/real_time/surface_src/java_src/Alice/logs/2016-02-20-06-50-45/webcam.mp4\"}");
+        writer.close();
+    }
+
+
+    /**
+     * Tests basic get/set functionality for a normal config file
+     * Doesn't actually create any robots
+     */
+    @Test
+    public void testRobotTypeGetSet() {
+
+        //Create a new config file with a reasonable robot type
+        try {
+            createConfigFileWithRobot("TransistorAuton");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            assert false;
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            assert false;
+        }
+
+        //Test getRobotType and setRobotType
+        assert RobobuggyConfigFile.getRobotType().equals("TransistorAuton");
+        RobobuggyConfigFile.setRobotType("TempBuggy!!!");
+        assert RobobuggyConfigFile.getRobotType().equals("TempBuggy!!!");
+    }
+
+    /**
+     * Creates a config file that specifies a TransistorAuton robot, then creates the robot
+     */
+    @Test
+    public void testTransistorAuton(){
+        try {
+            createConfigFileWithRobot("TransistorAuton");
+            RobobuggyConfigFile.loadConfigFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+            assert false;
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
+            assert false;
+        }
+        assert RobobuggyConfigFile.getRobot() instanceof TransistorAuton;
+    }
+
+    /**
+     * Creates a config file that specifies a TransistorDataCollection robot, then creates the robot
+     */
+    @Test
+    public void testTransistorDataCollection(){
```

Comment:
I don't think we need a separate test case for each robot. Simply looping through a list of them, reloading config each time would do it better